plugins/inputs/mikrotik/tools.go

```diff
@@ -0,0 +1,73 @@
+package mikrotik
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func createPropList() string {
+	var propList []string
+	var commonData CommonData
+	t := reflect.TypeOf(commonData)
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		jsonTag := field.Tag.Get("json")
+		if jsonTag != "" {
+			propList = append(propList, strings.Split(jsonTag, ",")[0])
+		}
+	}
+
+	return ".proplist=%" + strings.Join(propList, ",")
+}
+
+func binToCommon(b []byte) (c Common, err error) {
+	errCommon := json.Unmarshal(b, &c)
+	if errCommon == nil {
+		return c, nil
+	}
+
+	cd := CommonData{}
+	err = json.Unmarshal(b, &cd)
+
+	if err != nil {
+		return c, fmt.Errorf("data could not be unmarshalled neither into Common nor into CommonData structures. %w %w", errCommon, err)
+	}
+
+	c = append(c, cd)
+
+	return c, err
+}
+
+func basicCommentAndDisableFilter(commentsToIgnore []string) func(CommonData) bool {
+	return func(c CommonData) bool {
+		if c.Disabled == "true" {
+			return false
+		}
+		for _, comment := range commentsToIgnore {
+			if strings.Contains(c.Comment, comment) {
+				return false
+			}
+		}
+
+		return true
+	}
+}
+
+func filterCommon(data Common, f func(CommonData) bool) Common {
+	for i := 0; i < len(data); {
+		if !f(data[i]) {
+			data = append(data[:i], data[i+1:]...)
+		} else {
+			i++
+		}
+	}
+
+	return data
+}
```

Comment:
1. This is not an abstraction, it is redirection.
2. Then please change the code _in the future_ and reintroduce this! 

We already do have enough code that complicates things just because it might have been helpful in the future where this future never happened! So please tailor your plugin to what it does __now__ and extend it in the future if you must!